### DISCUSSION TITLE
Netbox Permissions After Upgrades

### DISCUSSION THREAD

**Discussion by cslingerland** [netbox-community/netbox#9298] [Q&A] [answered]
Whenever we upgrade to a new version of Netbox where new Models are introduced, the corresponding permissions don't seem to be created. I can look at the source and create them appropriately but it is a rather annoying headache. Am I doing something wrong in the upgrade process or is this expected behavior?

**Comment by mtinberg** [comment, marked as answer]
I have the same issues but I believe this is intentional as there is no way for the upgrade script to reliably know what Permissions entries you want the new Object Types on, so someone who has admin access to the instance has to make a decision.  You should be able to go into the Admin interface and add the Object Types to your existing Permissions after the upgrade, it's pretty straightforward to just CTRL-Click all the ones you want to add if you have a few Permissions entries that control a lot of Object Types.  If you've upgraded from some older releases you will end up with one Permission entry _per_ object type as part of a previous schema migration, so consolidating that first to a few named Permissions can be helpful.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison
